test/index.js

```diff
@@ -29,76 +18,86 @@ const describe = lab.describe;
+        const plugin = {
+            register: require('../lib/index').register,
+            options: {
+                reporters: [{
+                    reporter: GoodReporter,
+                    events: { response: '*' }
+                }]
+            }
         };
+        const server = new Hapi.Server();
+        server.register(plugin, (err) => {
 
-        server.connection({ tls: tlsOptions });
-        server.connection({ port: 0, labels: ['http'] });
+            expect(err).to.not.exist();
+            expect(server.plugins.good.monitor).to.be.an.object();
+            done();
+        });
+    });
 
-        server.route({
-            method: 'GET',
-            path: '/',
-            handler: function (request, reply) {
+    it('starts the Monitor object during registration', (done) => {
 
-                reply('ok');
+        const plugin = {
+            register: require('../lib/index').register,
+            options: {
+                reporters: [{
+                    reporter: GoodReporter,
+                    events: { response: '*' }
+                }]
             }
-        });
+        };
+        const server = new Hapi.Server();
+        let called = false;
+        const start = Monitor.prototype.start;
+        Monitor.prototype.start = (callback) => {
 
-        server.select('http').route({
-            method: 'GET',
-            path: '/http',
-            handler: function (request, reply) {
+            Monitor.prototype.start = start;
+            called = true;
+            callback();
+        };
 
-                reply('ok');
-            }
+        server.register(plugin, (err) => {
+
+            expect(err).to.not.exist();
+            expect(called).to.be.true();
+            done();
         });
+    });
 
-        let emitted;
-        const options = {};
-        options.reporters = [new GoodReporter({
-            wreck: '*'
-        }, null, (event) => {
-
-            if (!emitted) {
-                expect(event.event).to.equal('wreck');
-                expect(event.request.protocol).to.equal('https:');
-                expect(event.request.host).to.exist();
-                expect(event.request.path).to.equal('/');
-                emitted = true;
-            }
-            else {
-                expect(event.event).to.equal('wreck');
-                expect(event.request.protocol).to.equal('http:');
-                expect(event.request.host).to.exist();
-                expect(event.request.path).to.equal('/http');
-                done();
-            }
-        })];
+    it('stops the monitor when the server stops', (done) => {
 
         const plugin = {
-            register: require('..'),
-            options: options
+            register: require('../lib/index').register,
```

Comment:
don't need index... might want to add this at the top of the file as

`const Good = require('../')`